src/rules/noSubmoduleImportsRule.ts

```diff
@@ -0,0 +1,106 @@
+/**
+ * @license
+ * Copyright 2016 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+    isCallExpression,
+    isExternalModuleReference,
+    isIdentifier,
+    isImportDeclaration,
+    isImportEqualsDeclaration,
+    isTextualLiteral,
+} from "tsutils";
+import * as ts from "typescript";
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "no-submodule-imports",
+        description: Lint.Utils.dedent`
+            Disallows importing any submodule of the listed modules.`,
+        rationale: Lint.Utils.dedent`
+            Submodules of some packages are treated as private APIs and the import
+            paths may change without deprecation periods. It's best to stick with
+            top-level package exports.`,
+        optionsDescription: "A list of packages whose submodules are blacklisted.",
```

Comment:
And just to clarify: you mean that `import { forOwn } from "lodash";` and `import { Something } from "@angular/core";` would both be allowed by default, right?